scripts/pylint_extensions_test.py

```diff
@@ -161,10 +210,325 @@ def test(test_var_one, test_var_two): #@
+                msg_id='missing-yield-doc',
+                node=func_node
+            ), testutils.Message(
+                msg_id='missing-yield-type-doc',
+                node=func_node
+            ),
+        ):
+            checker_test_object.checker.visit_yieldfrom(yield_node)
+        with checker_test_object.assertNoMessages():
+            checker_test_object.checker.visit_return(yield_node)
+
+        func_node, return_node = astroid.extract_node("""
+        class Test(object):
+            def __init__(self, test_var_one, test_var_two): #@
+                \"\"\"Function to test docstring parameters.
+
+                Args:
+                    test_var_one: int. First test variable.
+                    test_var_two: str. Second test variable.
+
+                Yields:
+                    int. The test result.
+                \"\"\"
+                result = test_var_one + test_var_two
+                return result #@
+        """)
+        with checker_test_object.assertAddsMessages(
+            testutils.Message(
+                msg_id='redundant-yields-doc',
+                node=func_node
+            ),
+        ):
+            checker_test_object.checker.visit_functiondef(func_node)
+        with checker_test_object.assertAddsMessages(
+            testutils.Message(
+                msg_id='missing-return-doc',
+                node=func_node
+            ), testutils.Message(
+                msg_id='missing-return-type-doc',
+                node=func_node
+            ),
+        ):
+            checker_test_object.checker.visit_return(return_node)
+        with checker_test_object.assertNoMessages():
+            checker_test_object.checker.visit_yield(return_node)
+
+        func_node, raise_node = astroid.extract_node("""
+        def func(test_var_one, test_var_two): #@
+            \"\"\"Function to test docstring parameters.
+
+            Args:
+                test_var_one: int. First test variable.
+                test_var_two: str. Second test variable.
+
+            Raises:
+                Exception: An exception.
+            \"\"\"
+            raise Exception #@
+        """)
+        with checker_test_object.assertNoMessages():
+            checker_test_object.checker.visit_raise(raise_node)
+
+        func_node, raise_node = astroid.extract_node("""
+        def func(test_var_one, test_var_two): #@
+            \"\"\"Function to test raising exceptions.
+
+            Args:
+                test_var_one: int. First test variable.
+                test_var_two: str. Second test variable.
+            \"\"\"
+            raise Exception #@
+        """)
+        with checker_test_object.assertAddsMessages(
+            testutils.Message(
+                msg_id='missing-raises-doc',
+                args=('Exception',),
+                node=func_node
+            ),
+        ):
+            checker_test_object.checker.visit_raise(raise_node)
+
+        raise_node = astroid.extract_node("""
```

Comment:
Here and below: please chose a suitable naming convention for different `raise_node`.

Reply:
Done